mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManager.java

```diff
@@ -34,10 +34,27 @@
 /**
  * A component to manage the states of {@link TaskExecutorState} for a given {@link ResourceClusterActor}.
  */
-public interface ExecutorStateManager {
-    void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state);
-    void markAvailable(TaskExecutorID taskExecutorID);
-    void markUnavailable(TaskExecutorID taskExecutorID);
+interface ExecutorStateManager {
+    /**
+     * Store and track the given task executor's state inside this {@link ExecutorStateManager} if there is no existing
+     * state already. Ignore the given state instance if there is already a state associated with the given ID.
+     * @param taskExecutorID TaskExecutorID
+     * @param state new task executor state
+     */
+    void trackIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state);
+
+    /**
+     * Try to mark the given task executor as available if its tracked state is available.
+     * @param taskExecutorID TaskExecutorID
+     * @return whether the given task executor becomes available.
+     */
+    boolean tryMarkAvailable(TaskExecutorID taskExecutorID);
```

Comment:
If you look at TaskExecutorState and its use, it looks like this class is designed as a listener entity. Hence, the methods do not return any output and are designed as effects.